src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -732,6 +732,14 @@ message switch
                         var node = AutomationNode.GetOrCreate(peer);
                         return UiaCoreProviderApi.UiaReturnRawElementProvider(_hwnd, wParam, lParam, node);
                     }
+                    break;
+                case WindowsMessage.WM_SETTINGCHANGE:
+                    if (Marshal.PtrToStringAuto(lParam) == "ImmersiveColorSet"
+                        && Win32Platform.PlatformSettings is Win32PlatformSettings win32PlatformSettings)
+                    {
+                        win32PlatformSettings.OnColorValuesChanged(_hwnd);
```

Comment:
I wonder if this would be better placed on the Dispatcher's WndProc (in Win32Platform.cs) for cases where multiple windows are open, since this just passes the notification over to the platform settings. Also from my tests implementing this, WM_SETTINGCHANGE is sent at least 4 times with "ImmersiveColorSet" for a system setting change (not sure why). Even though Win32PlatformSettings does a check to not raise the event unless the values have changed, that still would amount to quite a few COM calls for UISettings, especially if multiple windows are open. 

Also, is there a reason we're passing the HWND into `OnColorValuesChanged`. That method doesn't seem to use that param

Reply:
> Also, is there a reason we're passing the HWND into OnColorValuesChanged. That method doesn't seem to use that param

No, there is no reason for that anymore. Leftover from previous experiments.

> I wonder if this would be better placed on the Dispatcher's WndProc (in Win32Platform.cs) 

Good idea, I will try that.

> Also from my tests implementing this, WM_SETTINGCHANGE is sent at least 4 times with "ImmersiveColorSet"

Yeah, noticed that. Same with iOS too which can really spam with changes. Not much we can do.

Reply:
I am also going to add simple HighContrast support in the low level API as @hez2010 suggested. Though there is no plans to support it on the framework level in themes (at least not in 11.0).
Maybe third-party themes will decide to use it before.

Although, only Win, macOS and Browser (on Win only I think) do support it